src/Microsoft.ML.AutoML/TrainerExtensions/SweepableParams.cs

```diff
@@ -124,7 +124,7 @@ public static IEnumerable<SweepableParam> BuildMatrixFactorizationParams()
                 new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.LearningRate), new object[] { 0.001f, 0.01f, 0.1f }),
                 new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.ApproximationRank), new object[] { 8, 16, 64, 128 }),
                 new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.Lambda), new object[] { 0.01f, 0.05f, 0.1f, 0.5f, 1f }),
-                new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.LossFunction), new object[] { MatrixFactorizationTrainer.LossFunctionType.SquareLossRegression, MatrixFactorizationTrainer.LossFunctionType.SquareLossOneClass }),
+                new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.LossFunction), new object[] { "MatrixFactorizationTrainer.LossFunctionType.SquareLossRegression", "MatrixFactorizationTrainer.LossFunctionType.SquareLossOneClass" }),
```

Comment:
can you do a nameof() on those?  It's a bit odd design that I don't think I fully understand why it is done this way but otherwise the fix looks reasonable to me, the whole point of the code that is currently crashing is to convert these to string anyway.

Reply:
Thanks for pointing out, it will cause bug when sweeping parameter, but it doesn't crash pipeline.